docs/explanation/lifecycle.rst

```diff
@@ -2,7 +2,62 @@
 Lifecycle details
 *****************
 
+Each part is built in five separate steps:
+
+#. ``PULL`` — The components for the part are retrieved from their stated
```

Comment:
afaik the PULL step includes both the fetching of the package dependencies and the fetching of the source, if relevant, right? So it might be a good idea to add that here (that this is where the part's `source` is fetched)

Reply:
I've clarified this — have a read and LMK what you think